src/dmd/mtype.d

```diff
@@ -9224,7 +9228,7 @@ const(char*)[2] toAutoQualChars(Type t1, Type t2)
 {
     auto s1 = t1.toChars();
     auto s2 = t2.toChars();
-    if (strcmp(s1, s2) == 0)
+    if (!t1.equals(t2) && strcmp(s1, s2) == 0)
```

Comment:
when would the type strings be the same but they are different types?

Reply:
When the strings aren't the full qualified type, e.g. `S` vs `tem!().S`.

Reply:
Please add comment to that effect.

Reply:
done.